modules/ROOT/pages/traversal-framework/traversal_framework_example.adoc

```diff
@@ -0,0 +1,211 @@
+:description: A Neo4j Traversal Framework example.
+
+:org-neo4j-graphdb-Direction-both: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Direction.html#BOTH
+
+[[examples-how-to-use-the-traversal-framework]]
+= Traversal Framework Example
+
+A link:{org-neo4j-graphdb-traversal-TraversalDescription}[`org.neo4j.graphdb.traversal.TraversalDescription`^] is built using a fluent interface.
+Such a description can then spawn several link:{org-neo4j-graphdb-traversal-Traverser}[`org.neo4j.graphdb.traversal.Traverser`^] objects.
+
+image::traversal_framework_example.svg[role="middle"]
+
+[NOTE]
+====
+The source code for the examples can be found at: link:https://github.com/neo4j/neo4j-documentation/blob/{neo4j-version}/embedded-examples/src/main/java/org/neo4j/examples/TraversalExample.java[`TraversalExample.java`^].
+====
+
+With the definition of the `RelationshipTypes` as
```

Comment:
I think a preamble setting up the example would be really useful. It seems to dive straight into details assuming we know we are looking for people Joe likes / knows in some graph.